### DISCUSSION TITLE
Best practices for sharing components between apps

### DISCUSSION THREAD

**Discussion by ya2s** [vercel/turborepo#6309] [Help]
I manage multiple front-end apps with turborepo.
Common components like the design system are managed under package/ui and referenced from multiple apps.
On the other hand, I don't know the best practice when referring to a component placed under an app from another app that includes business logic.

e.g.
apps/appA/shared
apps/appB/shared
When you have a directory structure like the one above, you may want to use components under shared from appA to appB (and vice versa).

Is there an optimal method in this case?

**Comment by mehulkar** [comment]
Check out this guide: https://turbo.build/repo/docs/handbook/sharing-code/internal-packages. I would not recommend trying to share across apps, rather you want to move shared components into a _real_ shared package that can be a dependency in both `appA` and `appB`.
